### PR TITLE
Zhessler add dsql auth token generator

### PR DESCRIPTION
**Pull request by Velfi**
## Description

This adds an auth token generator that works like the existing RDS auth token generator, but for a new service: DSQL.

To generate the DSQL SDK, run this command from the root of the project:

```
./gradlew :aws:sdk:assemble -Paws.services=+sts,+sso,+ssooidc,+dsql
```

Then, to run the auth token generator tests:

```sh
cd aws/sdk/build/aws-sdk/sdk/dsql
cargo test signing_works --all-features
```

## Testing

tests are included in the inlineable

## Checklist

- [x] For changes to the AWS SDK, generated SDK code, or SDK runtime crates, I have created a changelog entry Markdown file in the `.changelog` directory, specifying "aws-sdk-rust" in the `applies_to` key.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: .changelog/1733238900.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+applies_to: ["client"]
```
Comment (ysaito1001) [inline .changelog/1733238900.md:2]:
```suggestion
applies_to: ["aws-sdk-rust"]
```

File: aws/rust-runtime/aws-inlineable/src/dsql_auth_token.rs [hunk 2]
```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! Code related to creating signed URLs for logging in to Xanadu.
```
Comment (ysaito1001) [inline aws/rust-runtime/aws-inlineable/src/dsql_auth_token.rs:6]: Just to check, are we allowed to mention `Xanadu` in a public GitHub repo?

**Review by ysaito1001 — APPROVED** [review]
For server codegen CI failure, I think it'll go away if we just re-run failed job.

File: aws/rust-runtime/aws-inlineable/src/dsql_auth_token.rs (hunk 2 above)
Comment (Velfi) [reply aws/rust-runtime/aws-inlineable/src/dsql_auth_token.rs:6]: I'll update this.

**Review by drganjoo — APPROVED** [review]

**Pull request merged**